src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,209 @@
+import com.conveyal.datatools.manager.controllers.api.DeploymentController;
+import com.conveyal.datatools.manager.models.Deployment;
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.models.OtpServer;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Auto deploy new feed version to OTP server if {@link Project#autoDeploy} is enabled and other conditions are met
+ * (e.g., feed version has no critical errors, active deployment is not in progress, there are no other feed fetches in
+ * progress, etc.). This job must run after {@link ValidateFeedJob} as it has a dependency on the outcome of
+ * {@link FeedVersion#hasCriticalErrors}.
+ */
+public class AutoDeployJob extends MonitorableJob {
+    public static final Logger LOG = LoggerFactory.getLogger(AutoDeployJob.class);
+
+    /**
+     * Projects to be considered for auto deployment.
+     */
+    private List<Project> projects = new ArrayList<>();
+
+    /**
+     * A set of projects which have been locked by a instance of {@link AutoDeployJob} to prevent repeat
+     * deployments.
+     */
+    private static final Set<String> lockedProjects = Collections.synchronizedSet(new HashSet<>());
+
+    /**
+     * Auto deploy specific project.
+     */
+    public AutoDeployJob(Project project, Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects.add(project);
+    }
+
+    /**
+     * Auto deploy all projects.
+     */
+    public AutoDeployJob(Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects = Persistence.projects.getAll();
+    }
+
+    @Override
+    public void jobLogic() {
+        for (Project project : projects) {
+            Deployment deployment = Persistence.deployments.getById(project.pinnedDeploymentId);
+            // Define if project and deployment are candidates for auto deploy.
+            if (!project.autoDeploy ||
+                project.pinnedDeploymentId == null ||
+                deployment == null ||
+                lockedProjects.contains(project.id)) {
+                LOG.info("Project {} skipped for auto deployment as required criteria not met.", project.name);
+                continue;
+            }
+            try {
+                synchronized (lockedProjects) {
+                    if (!lockedProjects.contains(project.id)) {
+                        lockedProjects.add(project.id);
+                    } else {
+                        continue;
```

Comment:
Please add a `LOG.warn` here:

```suggestion
                        LOG.warn("Unable to acquire lock for project {}", project.name);
                        continue;
```